bvm/ballerina-runtime/src/main/java/org/ballerinalang/jvm/TypeChecker.java

```diff
@@ -1982,26 +1982,41 @@ private static boolean checkIsLikeTupleType(Object sourceValue, BTupleType targe
         }
 
         ArrayValue source = (ArrayValue) sourceValue;
-        if (source.size() != targetType.getTupleTypes().size()) {
+        if (targetType.getRestType() == null && source.size() != targetType.getTupleTypes().size()) {
             return false;
         }
 
+        List<BType> targetTypes = new ArrayList<>(targetType.getTupleTypes());
+        if (targetType.getRestType() != null) {
+            if (source.size() <= targetTypes.size()) {
+                return false;
+            } else {
+                int targetTypeSize = source.size() - targetTypes.size();
+                for (int i = 0; i < targetTypeSize; i++) {
+                    targetTypes.add(targetType.getRestType());
+                }
```

Comment:
Instead of doing this, how about just setting the bound size to the target type and then later, check the remainder of the source type members against the rest type?

Reply:
Changed